### ISSUE TITLE
Core Data Model - make Homology symmetric

### ISSUE THREAD

**Issue by sammyjava** [intermine/intermine#1599]
Similar to the discussion of synteny symmetry today, I think we should make the Homologue class symmetric w.r.t. its referenced genes. It's pretty wonky to have Homologue.gene and Homologue.homologue as currently in FlyMine. We should just have Homologue.genes, and those genes have a reference to that Homologue. So not much of a change: replace the two references with one collection in the new core data model.

**Comment by julie-sullivan** [comment]
@sammyjava Where would you put all the data that is in the Homologue object? Type, score, evidence, etc?

**Comment by sammyjava** [comment]
Those are still attributes. I'm just saying the Homologue. homologue, Homologue.gene thing is kind of silly. Homologue.genes makes more sense. No change otherwise.

**Comment by julie-sullivan** [comment]
But that's a one to many relationship @sammyjava. Scores and evidence are pair-specific.

**Comment by sammyjava** [comment]
Yeah, I'm just saying lets store the two homologous genes in a collection rather than two references. It's a minor tweak.

**Comment by julie-sullivan** [comment]
What do you gain from this change? It will be a lot of work to change the data model, even for a "minor tweak", and some mines won't change at all. 

Does the benefit outweigh the cost?

1. Friendly mine code update.
2. InterMine homologue data sources (homologene, treefam, ensembl, orthodb, panther etc)
3. Mine-specific homologue data sources and tests to be updated
4. update templates and queries 
   - what if users have saved templates and queries
5. Discussion and update with devs
6. Re-train users to new model
7. Update other tools ,e.g. GO tool, that use homologues
8. Extra (code) complexity because some mines will have this change and some will not
9. Rebuild database with new data model. Test and verify correct.

**Comment by sammyjava** [comment]
Probably not, given the installed base. It was just a suggestion to fix what I think is a dorky original data model. :) Since we decided to make SyntenyBlock symmetric w.r.t. SyntenicRegion it seemes appropriate to do the same for Homologue. My role is to throw wrenches into the works, your role is to deflect them before they break the works. :)

**Comment by sammyjava** [comment]
And I will add that because of the homologue/gene asymmetry you wind up with duplicate Homologue entries in some mines, doubling the size of what already is a gigantic amount of data. Example: http://intermine.modencode.org/preview/begin.do -- when I loaded that into Neo4j I discovered that there are two Homologue entries for every "physical" homology pair, switching which is the homologue and which is the gene. Same attributes of course since it's the same Homologue. So that's bad design.
